app/code/Bulbulatory/Recomendations/Controller/Adminhtml/Recommendation/Delete.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Bulbulatory\Recomendations\Controller\Adminhtml\Recommendation;
+
+use Magento\Backend\App\Action\Context;
+use Bulbulatory\Recomendations\Api\RecommendationRepositoryInterface;
+
+class Delete extends \Magento\Backend\App\Action
+{
+    protected $context;
+    protected $recommendationRepository;
+
+    public function __construct(
+        Context $context,
+        RecommendationRepositoryInterface $recommendationRepository
+    )
+    {
+        parent::__construct($context);
+        $this->recommendationRepository = $recommendationRepository;
+    }
+
+    public function execute()
+    {
+
+        $id = $this->getRequest()->getParam('recommendation_id');
+        $resultRedirect = $this->resultRedirectFactory->create();
+        
+        if ($id) {
+            try {
+                $recommendation = $this->recommendationRepository->getById($id);
+                $this->recommendationRepository->delete($recommendation);
+                $this->messageManager->addSuccessMessage(__('The recommendation has been deleted.'));
+            } catch (LocalizedException $exception) {
+                $this->messageManager->addErrorMessage(__('Cannot delete recommendation.'));
+            }
```

Comment:
Z tych trzech linii powyżej tylko `->getById($id)` rzuca wyjątek i jest to `NoSuchEntityException`, więc wyłapywanie `LocalizedException` nie ma większego sensu, ponieważ nigdy on tutaj nie wystąpi.

Dobrze byłoby w wiadomości do użytkownika podać np. ID rekomendacji, której nie można usunąć oraz zalogować ten wyjątek (`LoggerInterface`), aby w przypadku błędów ułatwić debugowanie.